File: pi_estimate.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main()
{
    ll num_points;
    cout << "Enter number of random points: ";
    cin >> num_points;

    ll circlepoints = 0;
    double x, y;

    

    for (ll i = 0; i < num_points; i++)
    {
        // Generate random numbers between -1 and 1
        x = (double)rand() / RAND_MAX * 2.0 - 1.0;
        y = (double)rand() / RAND_MAX * 2.0 - 1.0;

        // Check if the point is inside the circle
        if (x * x + y * y <= 1)
        {
            circlepoints++;
        }
    }

    double pi = 4.0 * ((double)circlepoints / num_points);

    cout << "Total Points: " << num_points << endl;
    cout << "Points inside circle: " << circlepoints << endl;
    cout << "Estimated Pi = " << pi << endl;

    return 0;
}

File: pi_estimate.py
import numpy as np

def main():
    num_points = int(input("Enter number of random points: "))
    
    # Generate all random points 
    points = np.random.uniform(-1.0, 1.0, (num_points, 2))
    
    # Calculate distances from origin for all points
    distances = np.sum(points**2, axis=1)
    
    # Count points inside circle
    circle_points = np.sum(distances <= 1)
    
    pi_estimate = 4.0 * (circle_points / num_points)
    
    print(f"Total Points: {num_points}")
    print(f"Points inside circle: {circle_points}")
    print(f"Estimated Pi = {pi_estimate}")

if __name__ == "__main__":
    main()
